syft/pkg/cataloger/rpm/file_cataloger.go

```diff
@@ -0,0 +1,141 @@
+package rpm
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	rpmdb "github.com/knqyf263/go-rpmdb/pkg"
+	"github.com/sassoftware/go-rpmutils"
+
+	"github.com/anchore/syft/internal"
+	"github.com/anchore/syft/syft/artifact"
+	"github.com/anchore/syft/syft/file"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/source"
+)
+
+type FileCataloger struct{}
+
+// NewFileCataloger returns a new RPM file cataloger object.
+func NewFileCataloger() *FileCataloger {
+	return &FileCataloger{}
+}
+
+// Name returns a string that uniquely describes a cataloger
+func (c *FileCataloger) Name() string {
+	return "rpm-file-cataloger"
+}
+
+// UsesExternalSources indicates that the rpm file cataloger does not use external sources
+func (c *FileCataloger) UsesExternalSources() bool {
```

Comment:
@spiffcs Did we revert this interface change?

Reply:
`UsesExternalSources` is part of the `Cataloger` interface: https://github.com/anchore/syft/blob/main/syft/pkg/cataloger/cataloger.go#L45